### PR TITLE
Guessing the secret formula

### PR DESCRIPTION
**Pull request by rlisahuang**
#110 

Secrets should be written in the following format:

~~~
```z3-duo
# starter code for the user
------ # six dashes
# secret hidden from the user (solution)
```
~~~

I have created a test file `website/src/docs-smtlib/07 - games/01 - test.md` for examples.

Note that new files added to the `07 - games` directory need to be also added in `sidebars/smtlibSidebars.js` (painful, I know).

Still work in progress, will fill Nikolaj's code snippet from #110 with proper logic later.

### CODE REVIEW

**Comment by rlisahuang** [comment]
@pelikhan should I merge to main with the barebone version for now so that @NikolajBjorner can start adding examples?

**Comment by pelikhan** [comment]
Yep
________________________________

**Comment by rlisahuang** [comment]
Will start a new PR for further implementations.

File: website/src/components/TutorialComponents/index.tsx [hunk 1]
```diff
@@ -164,8 +169,30 @@ export default function CustomCodeBlock(props: { input: CodeBlockProps }) {
...
+    let input = currCode;
+    if (isZ3Duo) {
+      input = `let user_input = ${currCode}
+      let secret_input = ${result.output}
+      ${z3DuoCode}`
+    }
+
     // `z3.interrupt` -- set the cancel status of an ongoing execution, potentially with a timeout (soft? hard? we should use hard)
-    runProcess(currCode)
+    runProcess(input)
```
Comment (pelikhan) [inline website/src/components/TutorialComponents/index.tsx:195]: Can we convert this code to async/await?

File: website/src/remark/render-code-blocks.mjs [hunk 2]
```diff
@@ -172,6 +172,7 @@ export default function plugin() {
             const editableRegex = /(always-editable)/;
             const alwaysEditable = editableRegex.test(meta);
             const lineNumRegex = /(show-line-numbers)/i;
+            const splitterRegex = /------/;
```
Comment (pelikhan) [inline website/src/remark/render-code-blocks.mjs:175]: Add start/end line?

File: website/src/remark/render-code-blocks.mjs [hunk 3]
```diff
@@ -193,15 +194,28 @@ export default function plugin() {
...
                         highlight: highlight,
                         statusCodes: {},
-                        code: value,
-                        result: {},
+                        code: code,
+                        result: result,
                         githubRepo: githubRepo,
-                        editable: false,
-                        readonly: true,
+                        editable: lang === 'z3-duo',
```
Comment (pelikhan) [inline website/src/remark/render-code-blocks.mjs:217]: You have a local for this

File: website/src/remark/render-code-blocks.mjs [hunk 4]
```diff
@@ -193,15 +194,28 @@ export default function plugin() {
...
+                        result = {output: solution};
+                    }
+
                     const val = JSON.stringify({
                         lang: lang,
                         highlight: highlight,
                         statusCodes: {},
-                        code: value,
-                        result: {},
+                        code: code,
```
Comment (pelikhan) [inline website/src/remark/render-code-blocks.mjs:214]: code:code -> code

**Review by pelikhan — APPROVED** [review]

File: website/src/components/TutorialComponents/index.tsx [hunk 5]
```diff
@@ -164,8 +169,30 @@ export default function CustomCodeBlock(props: { input: CodeBlockProps }) {
     let errorMsg;
 
     const runProcess = clientConfig[lang];
+    const z3DuoCode = `const s1 = new Z3.Solver()
```
Comment (pelikhan) [inline website/src/components/TutorialComponents/index.tsx:172]: Doesn't this have to be done in the react component as JavaScript?

File: website/src/components/TutorialComponents/index.tsx (hunk 1 above)
Comment (rlisahuang) [reply website/src/components/TutorialComponents/index.tsx:195]: so using try/catch and async/await?

File: website/src/remark/render-code-blocks.mjs (hunk 2 above)
Comment (rlisahuang) [reply website/src/remark/render-code-blocks.mjs:175]: line numbers?

**Pull request merged**